### PR TITLE
add dubbing agent & elevenlabs tools

### PR DESCRIPTION
**Pull request by 0xrohitgarg**
closes : https://github.com/video-db/Director/issues/36

### CODE REVIEW

File: backend/director/agents/dubbing.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                ELEVENLABS_API_KEY = os.getenv("ELEVENLABS_API_KEY")
+                if not ELEVENLABS_API_KEY:
+                    raise Exception("Elevenlabs API key not present in .env")
+                elevenlabs_tool = ElevenLabsTool(api_key=ELEVENLABS_API_KEY)
+                job_id = elevenlabs_tool.create_dub_job(
+                    source_url=download_response["download_url"],
+                    target_language=target_language_code,
+                )
+                self.output_message.actions.append(
+                    f"Dubbing job initiated successfully - Job ID: {job_id}"
```
Comment (ashish-spext) [inline backend/director/agents/dubbing.py:121]: Here we can simply say `Dubbing job initiated with Job ID: {job_id}"`
There is a slight chance that user might interpret as success in haste.

File: backend/director/agents/dubbing.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                )
+                self.output_message.push_update()
+
+                self.output_message.actions.append(
+                    "Waiting for dubbing process to complete..."
```
Comment (ashish-spext) [inline backend/director/agents/dubbing.py:126]: Let's have two `..` instead of three for consistency

File: backend/director/agents/dubbing.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                self.output_message.actions.append(
+                    f"Uploading dubbed video to VideoDB as 'Dubbed_{target_language} {video['name']}'"
+                )
+                self.output_message.push_update()
+
+            dubbed_video = self.videodb_tool.upload(
+                dubbed_file_path,
+                source_type="file_path",
+                media_type="video",
+                name=f"Dubbed_{target_language} {video['name']}",
```
Comment (ashish-spext) [inline backend/director/agents/dubbing.py:148]: Can we change it to either "{name} (Dubbed in {language)" or "(Dubbed) ({language}) {name}"?

**Review by ashish-spext — APPROVED** [review]
Looks good to me!

Tested on a video.

Just requested minor stylistic updates, let's merge it once done.

**Comment by ashish-spext** [comment]
Can elevenlabs give us a download URL instead of local file download?
If yes that can speed up the upload process (no need to download and then upload just give url)

File: backend/director/agents/dubbing.py (hunk 3 above)
Comment (0xrohitgarg) [reply backend/director/agents/dubbing.py]: Changed it to [Dubbed in {target_language}] {video['name']}

**Comment by 0xrohitgarg** [comment]
@ashish-spext , no option present to get download url on this endpoint [Get Dubbed File](https://elevenlabs.io/docs/api-reference/get-dubbed-file), 
Also no option present in [Dub a video or endpoint](https://elevenlabs.io/docs/api-reference/create-dub)

Notes for optimization for later:
- Try Dub a Audio endpoint, excepting that it can reduce elevenlabs processing time
- Try text to speech api endpoint

**Pull request merged**